### PR TITLE
Add Hotwire support for backend

### PR DESCRIPTION
**Pull request by kennyadsl**
**Description**

Add support for [stimulus-rails](https://github.com/hotwired/stimulus-rails) and [turbo-rails](https://github.com/hotwired/turbo-rails) in the backend.

It uses [import-maps](https://github.com/WICG/import-maps) and the plain assets pipeline to provide a smooth experience without the need of forcing users to depend on Webpacker or similar tools.

🚧 

This is just an experiment (yet very promising). The long-term goal here could be replacing the whole backbone/jQuery code with more modern stimulus controllers. Also, I think we can combine this with turbo, to provide a more hotwired experience.

**Not working with Rails 5.2**

Unfortunately, `stimulus-rails` only supports Rails 6.0+ so it won't work with Rails 5.2, which we still support.  We can both wait until Rails 5.2 EOL, or remove Rails 5.2 support starting from Solidus 3.0. I'm in favor of the latter if we think having stimulus is a good idea. 

**TODO:**

- [x] Test adding stimulus libraries
- [ ] Test overriding a stimulus controller from a host application if some customizations are needed
- [ ] Define a base application controller and make all the other controllers inherit from that one
- [ ] Test using [mixins](https://www.betterstimulus.com/architecture/mixins.html)
- [ ] Test adding turbo
- [ ] Test in a real application on a deployed environment (mainly to see if there are issues with assets paths)

**Checklist:**
- [ ] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [ ] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)

### CODE REVIEW

**Comment by peterberkenbosch** [comment]
Love this!

File: backend/app/assets/javascripts/controllers/spree/backend/navigation_controller.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Controller } from "stimulus"
+import stickybits from "spree/backend/stickybits"
+
+export default class extends Controller {
+  static targets = [ "header", "menu", "footer", "sticky" ]
```
[the remark this answers is not in the document]
Comment (kennyadsl) [reply backend/app/assets/javascripts/controllers/spree/backend/navigation_controller.js:5]: 🤔

[the remark this answers is not in the document]
Comment (kennyadsl) [reply backend/app/assets/javascripts/controllers/spree/backend/navigation_controller.js:5]: I think it's due to our `.eslintrc.json` that has `"es6": false,` now.

File: backend/app/assets/javascripts/controllers/spree/backend/hidden_by_checkbox_controller.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+  </div>
+
+ *
+ * Maybe not the best way to document usage, TODO: find a better (more standard) alternative.
+ */
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = [ "checkbox", "content" ]
```
[the remark this answers is not in the document]
Comment (peterberkenbosch) [reply backend/app/assets/javascripts/controllers/spree/backend/hidden_by_checkbox_controller.js:21]: This is correct usage, although I have seen this error message in another discussion. That was solved with using:

```
static get targets() {
  return ["checkbox", "content"];
}
```
I like the current way much better tbh, but wanted to share this.

[the remark this answers is not in the document]
Comment (peterberkenbosch) [reply backend/app/assets/javascripts/controllers/spree/backend/hidden_by_checkbox_controller.js:21]: some more background: https://stackoverflow.com/questions/60026651/safari-unexpected-token-expected-an-opening-before-a-methods-paramet

**Comment by denys-medynskyi** [comment]
hey @kennyadsl 

Rails 5.2 is [not supported anymore](https://endoflife.date/rails)

Can this PR get into work again?

P.S. I can takeover pull request

**Comment by jarednorman** [comment]
We can probably just close this in anticipation of the plans with the backend, right? (But keep the branch around for reference, as we may still want to use the work.)

**Pull request closed without merging**